src/Starlight/Draw/Radar.hs

```diff
@@ -107,22 +109,22 @@ targetBlipCount = 10
 shader :: Shader U V O
 shader = program $ \ u
   ->  vertex (\ V{ there, r, colour } IG{ colour2, sweep } -> main $ do
-    there <- let' "there" (there - here u)
+    there <- let' "there" (D.coerce there - D.coerce (here u))
     d     <- let' "d"     (D.norm there)
-    dir   <- let' "dir"   (there D.^* (1/D.norm there))
-    let perp v = vec2 (negate (v D.^.D._y)) (v D.^.D._x)
+    dir   <- let' "dir"   (there D.^* (1/d))
+    let perp v = vec2 [negate (v D.^.D._y), v D.^.D._x]
         angleOf vec = atan2' (vec D.^.D._y) (vec D.^.D._x)
         wrap mn mx x = ((x + mx) `mod'` (mx - mn)) + mn
-    edge  <- let' "edge"  (perp dir D.^* r D.^* 0.5 + dir D.^* d)
-    angle <- let' "angle" (angleOf there)
+    edge  <- let' "edge"  (perp dir D.^* D.coerce r D.^* 0.5 + there)
+    angle <- let' "angle" (D.coerce (angleOf there))
     radius <- var "radius" radius
-    let step = D.max' 1 (D.min' (get radius/float (fromIntegral targetBlipCount)) (d / 50))
+    let step = D.max' 1 (D.min' maxStep (D.coerce d / (50 / 1000))) -- FIXME: account for unit size without hard-coding conversion factor
     iff (gl_InstanceID `gt` 0)
       (radius .= step * float gl_InstanceID)
       (pure ())
-    minSweep <- let' "minSweep" (minBlipSize / (2 * pi * get radius))
-    sweep .= (minSweep `D.max'` abs (wrap (-pi) pi (angleOf edge - angle)))
-    pos   <- let' "pos"   (vec2 (cos angle) (sin angle) D.^* get radius)
+    minSweep <- let' "minSweep" (D.coerce (minBlipSize / (2 * pi * get radius)))
+    sweep .= (minSweep `D.max'` abs (wrap (-pi) pi (D.coerce (angleOf edge) - angle)))
+    pos   <- let' "pos"   (vec2 [cos angle, sin angle] D.^* D.coerce (get radius))
```

Comment:
There’s still some jumpiness in the radar, especially noticeable with target blips, when you get out to, say, Himalia. But it’s not super harmful to gameplay so I’m punting for now.